File: zorgmedewerker-angular/src/model/message.ts
export enum Sender {
    STAFF,
    PLATFORM
}

export enum Person {
    PATIENT,
    NONE
}

export enum Task {
    QUESTION_ID,
    QUESTION_TEXT,
    QUESTION_TIMESTAMP,

    MEAL,
    MEAL_ID,
    MEAL_NAME,
    MEAL_DESCRIPTION,
    MEAL_CALORIES,
    MEAL_ALLERGIES,
    MEAL_IMAGE,

    PATIENT,
    PATIENT_ID,
    PATIENT_NAME,
    PATIENT_BIRTHDATE,
    PATIENT_ALLERGIES,
}

export interface Message {
    sender: Sender;
    senderId: string;
    person: Person;
    personId: string;
    task: Task,
    taskId: string;
    data: string;
}

File: backend/src/main/java/com/pepper/backend/model/messaging/Message.java
package com.pepper.backend.model.messaging;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Message {

    private Sender sender;

    private String senderId;

    private Person person;

    private String personId;

    private Task task;

    private String taskId;

    private String data;

}
